mapper-processor/src/test/java/com/datastax/oss/driver/internal/mapper/processor/entity/EntityNamingStrategyTest.java

```diff
@@ -0,0 +1,131 @@
+        .addMethod(
+            MethodSpec.methodBuilder("setI")
+                .addParameter(TypeName.INT, "i")
+                .addModifiers(Modifier.PUBLIC)
+                .build())
+        .addMethod(
+            MethodSpec.methodBuilder("getI")
+                .returns(TypeName.INT)
+                .addModifiers(Modifier.PUBLIC)
+                .addStatement("return 0")
+                .build());
+  }
+
+  @Test
+  public void should_fail_if_both_convention_and_converter_specified() {
+    should_fail_with_expected_error(
+        "Invalid annotation configuration: "
+            + "NamingStrategy must have either a 'convention' or 'customConverterClass' argument, "
+            + "but not both",
+        "test",
+        CUSTOM_CONVERTER_CLASS,
+        entityTemplate()
+            .addAnnotation(
+                AnnotationSpec.builder(NamingStrategy.class)
+                    .addMember("convention", "$T.CASE_INSENSITIVE", NamingConvention.class)
+                    .addMember("customConverterClass", "$T.class", CUSTOM_CONVERTER_CLASS_NAME)
+                    .build())
+            .build());
+  }
+
+  @Test
+  public void should_fail_if_neither_convention_nor_converter_specified() {
+    should_fail_with_expected_error(
+        "Invalid annotation configuration: "
+            + "NamingStrategy must have either a 'convention' or 'customConverterClass' argument",
+        "test",
+        CUSTOM_CONVERTER_CLASS,
+        entityTemplate().addAnnotation(NamingStrategy.class).build());
+  }
+
+  @Test
+  public void should_warn_if_multiple_conventions_specified() {
+    should_succeed_with_expected_warning(
+        "Too many naming conventions: "
+            + "NamingStrategy must have at most one 'convention' argument "
+            + "(will use the first one: CASE_INSENSITIVE)",
+        "test",
+        CUSTOM_CONVERTER_CLASS,
+        entityTemplate()
+            .addAnnotation(
+                AnnotationSpec.builder(NamingStrategy.class)
+                    .addMember(
+                        "convention",
+                        "{ $1T.CASE_INSENSITIVE, $1T.EXACT_CASE }",
+                        NamingConvention.class)
+                    .build())
+            .build());
+  }
+
+  @Test
+  public void should_warn_if_multiple_converters_specified() {
+    should_succeed_with_expected_warning(
+        "Too many custom converters: "
+            + "NamingStrategy must have at most one 'customConverterClass' argument "
+            + "(will use the first one: test.CustomConverter)",
+        "test",
+        CUSTOM_CONVERTER_CLASS,
+        entityTemplate()
+            .addAnnotation(
+                AnnotationSpec.builder(NamingStrategy.class)
+                    .addMember(
+                        "customConverterClass",
+                        "{ $1T.class, $1T.class }",
+                        CUSTOM_CONVERTER_CLASS_NAME)
+                    .build())
+            .build());
+  }
```

Comment:
I think we should add a test when `@CqlName` is specified and `@NamingStrategy` and assert that warning was generated

Reply:
Mmm there is no warning when a column name is overridden.

Are you referring to [this comment](https://github.com/datastax/java-driver/pull/1236#discussion_r286427436)? If so, it generates an error that is covered at line 69.